lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -140,81 +183,66 @@ def test__has_access_descriptor(self):
         with self.assertRaises(ValueError):
             access._has_access_descriptor(user, 'not_load_or_staff', descriptor)
 
+    @ddt.data(
+        (True, None, False, access_response.VisibilityError),
+        (False, None, True),
+        (True, YESTERDAY, False, access_response.VisibilityError),
+        (False, YESTERDAY, True),
+        (True, TOMORROW, False, access_response.VisibilityError),
+        (False, TOMORROW, False, access_response.StartDateError)
```

Comment:
I think the `expected_error_type` field (last element of the tuple) doesn't really work with itertools